sympy/solvers/diophantine/diophantine.py

```diff
@@ -152,6 +136,223 @@ def __call__(self, *args):
         return self.subs(zip(self.parameters, args))
 
 
+class DiophantineEquationType:
+    """
+    Internal representation of a particular diophantine equation type.
+
+    Parameters
+    ==========
+
+    equation
+        The diophantine equation that is being solved.
+    free_symbols: list (optional)
+        The symbols being solved for.
+
+    Attributes
+    ==========
+
+    total_degree
+        The maximum of the degrees of all terms in the equation
+    homogeneous
+        Does the equation contain a term of degree 0
+    homogeneous_order
+        Does the equation contain any coefficient that is in the symbols being solved for
+    dimension
+        The number of symbols being solved for
+    """
+    name = None
+
+    def __init__(self, equation, free_symbols=None):
+        try:
+            if free_symbols is None:
+                free_symbols = list(equation.free_symbols)
+            assert free_symbols
+        except (AttributeError, AssertionError):
+            raise ValueError('equation should have 1 or more free symbols')
```

Comment:
I realise that you're just moving the code around here but I really don't like catching either `AttributeError` or `AssertionError` as these can hide bugs. Is it possible to remove them? Why is `AttributeError` being caught anyway?

Reply:
It's raised with the test `raises(ValueError, lambda: classify_diop(1))`. Probably it would be better to just sympify the argument.